### DISCUSSION TITLE
Buttons are dark

### DISCUSSION THREAD

**Discussion by captDaylight** [saadeghi/daisyui#1631] [Q&A] [answered]
Hi, I just installed DaisyUI and I feel like I might be missing something fundamental. I have some buttons like this:

```
<button type="button" class="btn btn-primary btn-lg">Primary</button>
<button type="button" class="btn btn-secondary btn-lg">Secondary</button>
<button type="button" class="btn btn-accent btn-lg">accent</button>
```

When they are rendered they look like this:
<img width="407" alt="Screen Shot 2023-02-22 at 1 39 48 PM" src="https://user-images.githubusercontent.com/700617/220726167-0cdd7f8e-c905-415e-bde6-60b6e5d84201.png">

and when I hover I can see the color:

<img width="383" alt="Screen Shot 2023-02-22 at 1 40 11 PM" src="https://user-images.githubusercontent.com/700617/220726345-354f4ebe-c098-4129-bff1-33536d12c9ea.png">

What am I missing to make the color show up without hovering?

**Comment by saadeghi** [comment]
Your buttons should look like this: https://play.tailwindcss.com/gOqWrW2GWE  
Otherwise, there is something wrong about your setup.  

1. Make sure the path to all your markup files are addressed in your `tailwind.config.js` file's `content` object.
2. Make sure there aren't any other CSS code or CSS package in your project that may apply unwanted styles to these buttons.

If you couldn't find the problem, please provide a reproduction repo so I can help finding the problem.

**Edit**: also if you're using packages like WindiCSS, UnoCSS, etc... please don't use them with daisyUI. These packages can not process Tailwind CSS plugins correctly. Use Tailwind CSS itself.

**Reply by captDaylight** [reply]
Hi @saadeghi thanks for the quick reply, I made a repro here https://github.com/captDaylight/daisy-repro

This video below should illustrate:
- the buttons wrapped as a SolidJS `Button` component result in dark buttons
- if I copy and drop in the resulting DOM from the rendered buttons back into my code, save, and then it all works
- when I remove it from my code and save, it maintains the colors...

https://user-images.githubusercontent.com/700617/220747081-231289bd-f765-4b0d-9220-9ec51bf957b8.mp4

**Comment by saadeghi** [comment, marked as answer]
In a Tailwind CSS project, you shouldn't use dynamic class names (like `btn-{variable}` or `bg-{variable}`) because when you use the string `primary` instead of `btn-primary` how could Tailwind know it should generate the `btn-primary` class?  
Read more: https://tailwindcss.com/docs/content-configuration#dynamic-class-names

If you really need to split the class names like that, you can [safelist all the needed class names](https://tailwindcss.com/docs/content-configuration#safelisting-classes) so Tailwind can generate them

**Reply by captDaylight** [reply]
Gotcha, thanks @saadeghi that makes sense!
